File: lib/core/constants/permissions.dart
class Permissions {
  const Permissions._();

  static const userCreate = 'user.create';
  static const userRead = 'user.read';
  static const userUpdate = 'user.update';
  static const userDelete = 'user.delete';
  static const roleManage = 'role.manage';
  static const settingsManage = 'settings.manage';
  static const auditRead = 'audit.read';
  static const dashboardRead = 'dashboard.read';
  static const profileManage = 'profile.manage';
  static const reportRead = 'report.read';
  static const analyticsRead = 'analytics.read';
}

class AppRoles {
  const AppRoles._();

  static const admin = 'ADMIN';
  static const shareholder = 'SHAREHOLDER';
  static const user = 'USER';
}

File: backend/app/api/v1/permissions.py
from fastapi import APIRouter, Depends

from app.api.v1.deps import require_permission

router = APIRouter()

DEFAULT_PERMISSIONS = [
    "user.create",
    "user.read",
    "user.update",
    "user.delete",
    "role.manage",
    "settings.manage",
    "audit.read",
    "dashboard.read",
    "profile.manage",
    "report.read",
    "analytics.read",
]


@router.get("", response_model=list[str])
def list_permissions(_=Depends(require_permission("role.manage"))) -> list[str]:
    return DEFAULT_PERMISSIONS
